### PR TITLE
fix: Resolve goto when the bridge is ready

### PR DESCRIPTION
**Pull request by doubleface**
Before we resolved the goto when the webview was reload but but it fails
when running goto twice in a row for example (and also in other cases on
slow devices).

Also added some enhancements on post-me logs display
- feat: Do not log non post-me messages in messenger
- fix: Call the listener when non null
- fix: Show requestId even when = 0
- feat: Better display of handshake logs
- feat: Hide log events

#### Checklist

Before merging this PR, the following things must have been done:

* [ ] Faithful integration of the mockups at all screen sizes
* [ ] Tested on iOS
* [X] Tested on Android
* [ ] Localized in English and French
* [ ] All changes have test coverage
* [ ] Updated README & CHANGELOG, if necessary

### CODE REVIEW

File: src/libs/ReactNativeLauncher.js [hunk 1]
```diff
@@ -410,6 +411,7 @@ class ReactNativeLauncher extends Launcher {
       throw new Error(`worker bridge restart init error: ${err.message}`)
     }
     this.emit('WORKER_RELOADED')
+    this.emit('worker:webview:ready')
   }
```
Comment (Crash--) [inline src/libs/ReactNativeLauncher.js:415]: > but it fails when running goto twice in a row for example 

Can you add a test for that please?

File: src/libs/ReactNativeLauncher.js [hunk 2]
```diff
@@ -309,6 +309,7 @@ class ReactNativeLauncher extends Launcher {
       timerId = setTimeout(() => {
         reject(ERRORS.SET_WORKER_STATE_TOO_LONG_TO_INIT)
       }, options?.timeout || SET_WORKER_STATE_TIMEOUT_MS)
+      this.emit('SET_WORKER_STATE', options)
```
Comment (Crash--) [inline src/libs/ReactNativeLauncher.js:312]: I don't understand why this change is necessary in this commit? Can you explain?

File: src/libs/bridge/ReactNativeLauncherMessenger.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ export default class ReactNativeLauncherMessenger extends MessengerInterface {
   }
```
Comment (Crash--) [inline src/libs/bridge/ReactNativeLauncherMessenger.js:15]: can you add the `why` in the commit message? What kind of messages are logged that we don't want anymore?

File: src/libs/bridge/ReactNativeLauncherMessenger.js [hunk 4]
```diff
@@ -41,7 +41,7 @@ export default class ReactNativeLauncherMessenger extends MessengerInterface {
    */
   onMessage(event) {
     const data = JSON.parse(event.nativeEvent.data)
-    this.listener({ data })
+    if (this.listener) this.listener({ data })
```
Comment (Crash--) [inline src/libs/bridge/ReactNativeLauncherMessenger.js:48]: I don't like this kind of code. 

`this.listener` is added during the `addMessageListener`. So we should not be able to pass within the `onMessage` function without calling `addMessageListener` first. 

So it's the argument we pass to the `addMessageListener` that is null. So we should not call the `addMessageListener` if the `listener` is null, no? Where are we calling this method?

File: src/libs/bridge/ReactNativeLauncherMessenger.js [hunk 5]
```diff
@@ -46,7 +46,7 @@ export default class ReactNativeLauncherMessenger extends MessengerInterface {
   onMessage(event) {
     const data = JSON.parse(event.nativeEvent.data)
     if (this.listener) this.listener({ data })
-    if (this.debug && data.type === '@post-me') {
+    if (this.debug && data.type === '@post-me' && data.eventName !== 'log') {
```
Comment (Crash--) [inline src/libs/bridge/ReactNativeLauncherMessenger.js:49]: This commit should be squashed with the one adding the `post-me` check. But I still don't understand the need.

File: src/libs/ReactNativeLauncher.js (hunk 1 above)
Comment (Crash--) [reply src/libs/ReactNativeLauncher.js:415]: Since this is at least the second time in a row we have an issue on this case, we should add a lost of tests!

File: src/libs/ReactNativeLauncher.js (hunk 2 above)
Comment (doubleface) [reply src/libs/ReactNativeLauncher.js:312]: Je vais mieux y arriver en français : l'évènement `SET_WORKER_STATE` est ce qui provoque potentiellement le rechargement de la webview et donc l'évènement `worker:webview:ready`. Avant ce changement, on écoutait l'évènement `worker:webview:ready` après avoir déclenché `SET_WORKER_STATE`, il y avait donc un risque (minime) de rater `worker:webview:ready` et de rester bloqué.

File: src/libs/bridge/ReactNativeLauncherMessenger.js (hunk 5 above)
Comment (doubleface) [reply src/libs/bridge/ReactNativeLauncherMessenger.js:49]: These are different : 
 * this one hides post-me message related to the transmission of console.logs in the webviews
 * the one adding post-me check hides postMessage messages not related to post-me

I don't think we should merge them.

Globally, the need here is to make log messages related to post-me easier to scan visually and easier to link, making it easier to find messages related to a given session or a given request.

File: src/libs/bridge/ReactNativeLauncherMessenger.js (hunk 4 above)
Comment (doubleface) [reply src/libs/bridge/ReactNativeLauncherMessenger.js:48]: As I say in this commit message, I only have this problem in scenarios when I watch the flagship app and with hot reload.

`addMessageListener` is called directly by post-me itself

File: src/libs/ReactNativeLauncher.js (hunk 1 above)
Comment (Crash--) [reply src/libs/ReactNativeLauncher.js:415]: You can just test the ReactNativeLauncher, without the LauncherView. We can test that the events are well emitted. 

But OK to do that later.

File: src/libs/bridge/ReactNativeLauncherMessenger.js (hunk 4 above)
Comment (Crash--) [reply src/libs/bridge/ReactNativeLauncherMessenger.js:48]: You should add this comment to the code directly. 👍
